closed/adds/jdk/src/share/classes/sun/security/provider/NativeDigest.java

```diff
@@ -49,13 +50,27 @@
     // length of the message digest in bytes
     private final int digestLength;
     private final int algIndx;
+    private Cleaner cleaner;
+
     // number of bytes processed so far. subclasses should not modify
     // this value.
     // also used as a flag to indicate reset status
     //  0: is already reset
     private long bytesProcessed;
 
-    private static NativeCrypto nativeCrypto;
+    private static final NativeCrypto nativeCrypto;
+
+    private static final class DigestCleanerRunnable implements Runnable {
+        private final long digestCtx;
+
+        public DigestCleanerRunnable(long context) {
+            this.digestCtx = context;
+        }
+
+        synchronized public void run() {
```

Comment:
Why is this method `synchronized`? It can't help because each context will have a distinct instance of `DigestCleanerRunnable`.

Reply:
removed